### PR TITLE
Improve install script

### PR DESCRIPTION
**Pull request by thiagokokada**
- Add `--checksum` option. This will allow to check the SHA256 sum of the download, making sure that we are downloading the correct binary. By default, there is no check, to keep this script backwards compatible (e.g.: if someone doesn't use `--checksum` option, they don't need to have `sha256sum` installed)
- Add `--static` option. This will, on Linux systems, download the static release instead of the dynamic one. On macOS it exit with error
- Some small refactorings

### CODE REVIEW

File: install [hunk 1]
```diff
@@ -25,14 +33,10 @@ if [[ $# -eq 1 ]]; then
    install_dir=${1:-}
 fi
 
-while [[ $# -gt 1 ]]
+while [[ $# -gt 0 ]]
 do
     key="$1"
-    if [[ -z "${2:-}" ]]; then
-        print_help
-    fi
```
Comment (thiagokokada) [inline install:33]: Those changes were necessary to support boolean options like `--static`. As a bonus, `--help` now works too (before it just ignored and started download with the default options).

File: install [hunk 2]
```diff
@@ -3,20 +3,28 @@
...
     echo -e
     echo "Usage:"
-    echo "install [--dir <dir>] [--download-dir <download-dir>] [--version <version>]"
+    echo "install [--dir <dir>] [--download-dir <download-dir>] [--version <version>] [--checksum <checksum>] [--static]"
     echo -e
     echo "Defaults:"
     echo " * Installation directory: ${default_install_dir}"
     echo " * Download directory: ${default_download_dir}"
+    if [[ -z "$checksum" ]]; then
+        echo " * Checksum: no"
```
Comment (borkdude) [inline install:23]: I don't believe we need this logging, do we?

Comment (thiagokokada) [reply install]: @borkdude Done.

Comment (borkdude) [reply install]: Oh sorry, I misread, this is the help output. 🤦

Comment (borkdude) [reply install]: Feel free to add it back, my bad.

**Pull request merged**